integration/run_test.go

```diff
@@ -185,3 +191,216 @@ func testWebServiceContainer(shell *shell.Shell, containerName string) {
+min_wait_msec = ` + strconv.FormatInt(tt.config.MinWaitMsec, 10) + `
+max_wait_msec = ` + strconv.FormatInt(tt.config.MaxWaitMsec, 10) + `
+
+[background_fetch]
+disable = true
+`
+
+			m := rebootContainerd(t, sh, getContainerdConfigToml(t, false), getSnapshotterConfigToml(t, false, config))
+			// Mirror image
+			copyImage(sh, dockerhub(containerImage), regConfig.mirror(containerImage))
+			// Pull image, create SOCI index with all layers and small (100kiB) spans
+			indexDigest := buildIndex(sh, regConfig.mirror(containerImage), withMinLayerSize(0), withSpanSize(100*1024))
+			sh.X("soci", "image", "rpull", "--user", regConfig.creds(), "--soci-index-digest", indexDigest, regConfig.mirror(containerImage).ref)
+
+			// Run the container
+			image := regConfig.mirror(containerImage).ref
+			sh.X("soci", "run", "-d", "--snapshotter=soci", image, "test-container")
+
+			sh.X("apt-get", "--no-install-recommends", "install", "-y", "iptables")
+
+			// TODO: Wait for the container to be up and running
+
+			// Block network access to the registry
+			if tt.config.NetworkDisableMsec > 0 {
+				sh.X("iptables", "-A", "OUTPUT", "-d", registryHostIP, "-j", "DROP")
+			}
+
+			// Do something in the container that should work without network access
+			commandSucceedStdout, _, err := sh.R("ctr", "task", "exec", "--exec-id", "test-task-1", "test-container", "sh", "-c", "times")
+			if err != nil {
+				t.Fatalf("attempt to run task without network access failed: %s", err)
+			}
+
+			type ErrorLogLine struct {
+				Error string `json:"error"`
+				Msg   string `json:"msg"`
+			}
+
+			gaveUpChannel := make(chan bool, 1)
+			defer close(gaveUpChannel)
+
+			monitorGaveUp := func(rawL string) {
+				if i := strings.Index(rawL, "{"); i > 0 {
+					rawL = rawL[i:] // trim garbage chars; expects "{...}"-styled JSON log
+				}
+				var logLine ErrorLogLine
+				if err := json.Unmarshal([]byte(rawL), &logLine); err == nil {
+					if logLine.Msg == "statFile error" && strings.Contains(logLine.Error, "giving up after") {
+						gaveUpChannel <- true
+						return
+					}
+				}
+			}
+
+			m.Add("retry", monitorGaveUp)
+			defer m.Remove("retry")
+
+			// Do something in the container to access un-fetched spans, requiring network access
+			commandNetworkStdout, _, err := sh.R("ctr", "task", "exec", "--exec-id", "test-task-2", "test-container", "cat", "/etc/hosts")
+			if err != nil {
+				t.Fatalf("attempt to run task requiring network access failed: %s", err)
+			}
+
+			// Wait with network disabled
+			time.Sleep(time.Duration(tt.config.NetworkDisableMsec) * time.Millisecond)
+
+			// Short wait to allow commands to complete
+			time.Sleep(time.Duration(1000) * time.Millisecond)
+
+			// Confirm first command succeeded while network was down
+			buf := make([]byte, 100)
+			_, err = commandSucceedStdout.Read(buf)
+			if err != nil {
+				t.Fatalf("read from expected successful task output failed: %s", err)
+			}
+			if !strings.Contains(string(buf), "s ") {
```

Comment:
what does this mean? (e.g. the `s ` string)

Reply:
`times` produces an output that looks something like this:
```
0m0.03s 0m0.05s
0m4.52s 0m1.98s
```
I could match more precisely with a more complex regex if you'd prefer?